tests/org.eclipse.ui.tests.dnd/src/org/eclipse/ui/tests/dnd/DragOperations.java

```diff
@@ -0,0 +1,92 @@
+package org.eclipse.ui.tests.dnd;
+
+import org.eclipse.jface.util.Geometry;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.IWorkbenchPartReference;
+import org.junit.Assert;
+
+public class DragOperations {
+
+	/**
+	 * Drags the given view OR editor to the given location (i.e. it only cares that we're given
+	 * a 'Part' and doesn't care whether it's a 'View' or an 'Editor'.
+	 * <p>
+	 * This method should eventually replace the original one once the Workbench has been updated
+	 * to handle Views and Editors without distincton.
+	 */
+	public static void drag(IWorkbenchPart part, TestDropLocation target, boolean wholeFolder) {
```

Comment:
I don't think the tests can really work of this essential method still fails, do they?